contracts/orderbook/Logic/OrderBook.sol

```diff
@@ -0,0 +1,425 @@
+                return false;
+            }
+            uint256 dSwapValue;
+            if (order.isCollateral) {
+                (dSwapValue, ) = swapImpl.dexAmountOutFormatted(
+                    payload,
+                    order.collateralTokenAmount
+                );
+            } else {
+                (dSwapValue, ) = swapImpl.dexAmountInFormatted(
+                    payload,
+                    order.collateralTokenAmount
+                );
+            }
+            if (order.amountReceived <= dSwapValue) {
+                return true;
+            }
+        } else {
+            if (!isActiveLoan(order.loanID)) {
+                return false;
+            }
+            bool operand;
+            if (_useOracle[trader]) {
+                operand =
+                    order.amountReceived >=
+                    queryRateReturn(
+                        order.base,
+                        order.loanTokenAddress,
+                        order.collateralTokenAmount
+                    ); //TODO: Adjust for precision
+            } else {
+                operand =
+                    order.amountReceived >=
+                    getDexRate(
+                        swapImpl,
+                        order.base,
+                        order.loanTokenAddress,
+                        payload,
+                        order.collateralTokenAmount
+                    );
+            }
+            if (operand) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function getDexRate(
+        ISwapsImpl swapImpl,
+        address base,
+        address loanTokenAddress,
+        bytes memory payload,
+        uint256 amountIn
+    ) public returns (uint256 rate) {
+        (rate, ) = swapImpl.dexAmountOutFormatted(
+            payload,
+            10**IERC20Metadata(base).decimals()
+        );
+        rate = (rate * amountIn) / 10**IERC20Metadata(base).decimals();
+    }
+
+    function priceCheck(
+        address loanTokenAddress,
+        address base,
+        ISwapsImpl swapImpl,
+        bytes memory payload
+    ) public returns (bool) {
+        uint256 dexRate = getDexRate(
+            swapImpl,
+            base,
+            loanTokenAddress,
+            payload,
+            10**IERC20Metadata(base).decimals()
+        );
+        uint256 indexRate = queryRateReturn(
+            base,
+            loanTokenAddress,
+            10**IERC20Metadata(base).decimals()
+        );
+        if (dexRate >= indexRate) {
+            if (((dexRate - indexRate) * 1000) / dexRate <= 5) {
+                return true;
+            } else {
+                return false;
+            }
+        } else {
+            if (((indexRate - dexRate) * 1000) / indexRate <= 5) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+    }
+
+    function executeOrder(bytes32 orderID) public {
```

Comment:
its a little bit big function maybe split it in separate things? if possible

Reply:
executeOrder is the function that is called to trigger an order. I can split the logic portion into three different functions internally as each if statement (once it is run through) automatically returns anyways. It wont change functionality but will make it easier to review